sphinx/builders/__init__.py

```diff
@@ -597,7 +598,7 @@ def write(
 
         if build_docnames is None or build_docnames == ['__all__']:
             # build_all
-            build_docnames = self.env.found_docs
+            build_docnames = self.env.all_docs
```

Comment:
Should it really be `all_docs` and not just `found_docs`? because `all_docs` also contain those that could have been removed.